src/policy_evaluator_builder.rs

```diff
@@ -45,6 +63,14 @@ impl PolicyEvaluatorBuilder {
         self
     }
 
+    /// Use a pre-built [`wasmtime::Module`] instance.
+    /// **Warning:** you must provide also the [`wasmtime::Engine`] used
+    /// to allocate the `Module`, otherwise the code will panic at runtime
```

Comment:
this will be checked in `validate_user_input`, and then it would return an error in `build`. I don't see how this can panic in the `policy-evaluator`. If it panics when it is called in the `policy-server` it might makes more sense to move the comment there

Reply:
I prefer to leave it there, because this is something any kind of consumer of `PolicyEvaluatorBuilder` must know.

The check inside of `validate_user_input` cannot catch all kind of misuses. I'm referring to something like:

* User creates a `wasmtime::Engine` called `eng1`, which has the "fuel" feature enabled
* User creates a `wasmtime::Module` using `eng1`. The module will be compiled to have the fuel instructions added
* User uses `PolicyEvaluatorBuilder` to create a `PolicyEvaluator`. At creation time the user provides the module created previously and an engine via `wasmtime::Engine::new`

In this scenario, `PolicyEvaluatorBuilder::validate_user_input` will spot 0 issues. But at runtime, the code is going to panic because the engine being used has a different configuration from the one that compiled the module